.github/workflows/rust.yml

```diff
@@ -6,13 +6,40 @@ on:
 
 jobs:
   build:
-    runs-on: ${{ matrix.os }}-latest
     strategy:
       matrix:
-        channel: [stable, beta, nightly]
-        os: [ubuntu, macos, windows]
+        type: [stable, beta, nightly, windows, macos]
+        include:
+          - type: stable
+            rust: stable
+            os: ubuntu-latest
+          - type: beta
+            rust: beta
+            os: ubuntu-latest
+          - type: nightly
+            rust: nightly
+            os: ubuntu-latest
+          - type: windows
+            rust: stable
+            os: windows-latest
+          - type: macos
+            rust: stable
+            os: macos-latest
+    runs-on: ${{ matrix.os }}
```

Comment:
Can you either revert the change to the strategy matrix or explain the change? It doesn't look like it adds anything to me.

Reply:
In fact it removes a tiny bit - I am open to doing it your way but this saves a little bit of processing power to only test beta and nightly on linux rather than all platforms, so shorter CI times and less runners used allowing this repo to scale better in the future at the cost of minor loss of information in case something goes wrong on these two platforms on the non-stable channel.

Reply:
Yeah, I want to test beta and nightly on macos and windows too. I don't care about CI times or less runners. It takes less than 5 minutes, and I'd prefer to be able catch a upcoming breaking change or regression in Rust.

Reply:
Alright, gonna revert it!